src/DotNetty.Transport/Channels/DefaultChannelId.cs

```diff
@@ -0,0 +1,419 @@
+                int.TryParse(customProcessId, out processId);
+            }
+            if (processId < 0 || processId > MaxProcessId)
+            {
+                processId = -1;
+            }
+            if (processId < 0)
+            {
+                processId = DefaultProcessId();
+            }
+            ProcessId = processId;
+            byte[] machineId = null;
+            string customMachineId = SystemPropertyUtil.Get("io.netty.machineId");
+            if (customMachineId != null)
+            {
+                if (MachineIdPattern.Match(customMachineId).Success)
+                {
+                    machineId = ParseMachineId(customMachineId);
+                }
+            }
+
+            if (machineId == null)
+            {
+                machineId = DefaultMachineId();
+            }
+            MachineId = machineId;
+        }
+
+        static byte[] ParseMachineId(string value)
+        {
+            // Strip separators.
+            value = value.Replace("[:-]", "");
+            byte[] machineId = new byte[MachineIdLen];
+            for (int i = 0; i < value.Length; i += 2)
+            {
+                machineId[i] = (byte)int.Parse(value.Substring(i, i + 2), NumberStyles.AllowHexSpecifier);
+            }
+            return machineId;
+        }
+
+        static int DefaultProcessId()
+        {
+            int pId = -1;
+            try
+            {
+                pId = Process.GetCurrentProcess().Id;
+            }
+            catch
+            {
+                // ignored
+            }
+
+            if (pId < 0)
+            {
+                pId = ThreadLocalRandom.Value.Next(MaxProcessId + 1);
+            }
+            return pId;
+        }
+
+        public static DefaultChannelId NewInstance()
+        {
+            var id = new DefaultChannelId();
+            id.Init();
+
+            return id;
+        }
+
+        static byte[] DefaultMachineId()
+        {
+            // Find the best MAC address available.
+            byte[] notFound = { byte.MaxValue };
+            byte[] bestMacAddr = notFound;
+            IPAddress bestIpAddr = IPAddress.Loopback;
+            var ifaces = new SortedDictionary<NetworkInterface, IPAddress>();
+            try
+            {
+                foreach (NetworkInterface iface in NetworkInterface.GetAllNetworkInterfaces())
+                {
+                    var addrs = iface.GetIPProperties().UnicastAddresses;
+                    var addr = addrs.FirstOrDefault(a => !IPAddress.IsLoopback(a.Address));
+                    if (addr != null)
+                    {
+                        ifaces.Add(iface, addr.Address);
+                    }
+                }
+            }
+                //catch (SocketException e)
+                //{
+                //    logger.warn("Failed to retrieve the list of available network interfaces", e);
+                //}
+            catch
+            {
+                // ignored
```

Comment:
`Logger.Warn`; should probably catch only `NetworkInformationException` per spec.